generators/server/templates/src/main/java/package/config/_CacheConfiguration.java

```diff
@@ -278,4 +313,212 @@ public SessionRegistry sessionRegistry() {
     }
     <%_ } _%>
     <%_ } _%>
+
+    <%_ if (hibernateCache === 'infinispan') { _%>
+    private final Logger log = LoggerFactory.getLogger(CacheConfiguration.class);
+
+    // to initialize cache in a non spring managed beans
+    private static EmbeddedCacheManager cacheManager;
+        <%_ if (serviceDiscoveryType === 'eureka') { _%>
+
+    private DiscoveryClient discoveryClient;
+
+    private Registration registration;
+
+    @Autowired(required = false)
+    public void setRegistration(Registration registration) {
+        this.registration = registration;
+    }
+
+    @Autowired(required = false)
+    public void setDiscoveryClient(DiscoveryClient discoveryClient) {
+        this.discoveryClient = discoveryClient;
+    }
+        <%_ } _%>
+
+    public static EmbeddedCacheManager getCacheManager(){
+        return cacheManager;
+    }
+
+    public static void setCacheManager(EmbeddedCacheManager cacheManager) {
+        CacheConfiguration.cacheManager = cacheManager;
+    }
+
+    /**
+     * <p>
```

Comment:
Note to self: I will rewrite this, so we have the same style as the rest (don't worry @srinivasa-vasu this is very minor)